posthog/api/insight.py

```diff
@@ -310,7 +313,11 @@ def _update_insight_dashboards(self, dashboards: List[Dashboard], instance: Insi
         for dashboard in candidate_dashboards:
             if dashboard.team != instance.team:
                 raise serializers.ValidationError("Dashboard not found")
-            DashboardTile.objects.create(insight=instance, dashboard=dashboard)
+            tile, _ = DashboardTile.objects.get_or_create(insight=instance, dashboard=dashboard)
```

Comment:
kind of confused, never touched this bit, what's the relation between a tile and an insight?

And what does a deleted tile mean?

Reply:
I think what has happened is that we've added a unique constraint to dashboard/tile in the DB when adding text cards. _After_ we wrote this code and the tests aren't forcing out where it is now fighting the DB.

Reply:
Thanks, that's very useful to know!